proxy/envoy/config.go

```diff
@@ -674,17 +674,21 @@ func buildInboundListeners(mesh *proxyconfig.MeshConfig, sidecar proxy.Node,
 				model.SortRouteRules(rules)
 				for _, config := range rules {
 					rule := config.Spec.(*proxyconfig.RouteRule)
+					var route *HTTPRoute
 					if rule.WebsocketUpgrade {
-						websocketRoute := buildInboundWebsocketRoute(rule, cluster)
-
+						route = buildInboundWebsocketRoute(rule, cluster)
+					} else if config.ConfigMeta.Name != "" {
+						route = buildInboundRoute(config, rule, cluster)
```

Comment:
why add another route and not change default route?

Reply:
I think I'm concerned that we rely on route rules by destination here. We cannot be certain that the request originates from within the mesh (or from within the snapshot of the rules known at destination). If a request doest not fit any rule, then this logic does not apply.

Reply:
My understanding is that there could be multiple routes here - if multiple routes are defined for different endpoints on the same destination? If so, then each endpoint should be configured with a different span/operation name.